Subject: [PATCH] CamcorderProfiles: Add new camcorder profiles

Below changes are squashed:

video: Add QHD & 2K profile for recorder
CamcorderProfiles: add VGA and 4K-DCI profiles
Also add high-speed profile for CIF.

diff --git a/media/libmedia/MediaProfiles.cpp b/media/libmedia/MediaProfiles.cpp
index aade69a09..f101c0438 100644
--- a/media/libmedia/MediaProfiles.cpp
+++ b/media/libmedia/MediaProfiles.cpp
@@ -91,6 +91,19 @@ const MediaProfiles::NameToTagMap MediaProfiles::sCamcorderQualityNameMap[] = {
     {"highspeed720p", CAMCORDER_QUALITY_HIGH_SPEED_720P},
     {"highspeed1080p", CAMCORDER_QUALITY_HIGH_SPEED_1080P},
     {"highspeed2160p", CAMCORDER_QUALITY_HIGH_SPEED_2160P},
+
+    // Vendor-specific profiles
+    {"vga", CAMCORDER_QUALITY_VGA},
+    {"4kdci", CAMCORDER_QUALITY_4KDCI},
+    {"timelapsevga", CAMCORDER_QUALITY_TIME_LAPSE_VGA},
+    {"timelapse4kdci", CAMCORDER_QUALITY_TIME_LAPSE_4KDCI},
+    {"highspeedcif", CAMCORDER_QUALITY_HIGH_SPEED_CIF},
+    {"highspeedvga", CAMCORDER_QUALITY_HIGH_SPEED_VGA},
+    {"highspeed4kdci", CAMCORDER_QUALITY_HIGH_SPEED_4KDCI},
+    {"qhd", CAMCORDER_QUALITY_QHD},
+    {"2k", CAMCORDER_QUALITY_2k},
+    {"timelapseqhd", CAMCORDER_QUALITY_TIME_LAPSE_QHD},
+    {"timelapse2k", CAMCORDER_QUALITY_TIME_LAPSE_2k},
 };
 
 #if LOG_NDEBUG
@@ -426,8 +439,10 @@ MediaProfiles::startElementHandler(void *userData, const char *name, const char
 }
 
 static bool isCamcorderProfile(camcorder_quality quality) {
-    return quality >= CAMCORDER_QUALITY_LIST_START &&
-           quality <= CAMCORDER_QUALITY_LIST_END;
+    return (quality >= CAMCORDER_QUALITY_LIST_START &&
+           quality <= CAMCORDER_QUALITY_LIST_END) ||
+           (quality >= CAMCORDER_QUALITY_VENDOR_START &&
+           quality <= CAMCORDER_QUALITY_VENDOR_END);
 }
 
 static bool isTimelapseProfile(camcorder_quality quality) {
diff --git a/media/libmedia/include/media/MediaProfiles.h b/media/libmedia/include/media/MediaProfiles.h
index 6975581f4..b3ab08dd8 100644
--- a/media/libmedia/include/media/MediaProfiles.h
+++ b/media/libmedia/include/media/MediaProfiles.h
@@ -56,6 +56,20 @@ enum camcorder_quality {
     CAMCORDER_QUALITY_HIGH_SPEED_1080P = 2004,
     CAMCORDER_QUALITY_HIGH_SPEED_2160P = 2005,
     CAMCORDER_QUALITY_HIGH_SPEED_LIST_END = 2005,
+
+    CAMCORDER_QUALITY_VENDOR_START = 10000,
+    CAMCORDER_QUALITY_VGA = 10000,
+    CAMCORDER_QUALITY_4KDCI = 10001,
+    CAMCORDER_QUALITY_TIME_LAPSE_VGA = 10002,
+    CAMCORDER_QUALITY_TIME_LAPSE_4KDCI = 10003,
+    CAMCORDER_QUALITY_HIGH_SPEED_CIF = 10004,
+    CAMCORDER_QUALITY_HIGH_SPEED_VGA = 10005,
+    CAMCORDER_QUALITY_HIGH_SPEED_4KDCI = 10006,
+    CAMCORDER_QUALITY_QHD = 10007,
+    CAMCORDER_QUALITY_2k = 10008,
+    CAMCORDER_QUALITY_TIME_LAPSE_QHD = 10009,
+    CAMCORDER_QUALITY_TIME_LAPSE_2k = 10010,
+    CAMCORDER_QUALITY_VENDOR_END = 10010,
 };
 
 enum video_decoder {
